addon/mixins/route.js

```diff
@@ -216,42 +122,56 @@ const RouteMixin = Mixin.create({
       delete options.storeFindMethod;
     }
 
-    const startingPage = options.startingPage === undefined ? 0 : options.startingPage-1;
-    const perPage      = options.perPage || this.get('_perPage');
-    const modelPath    = options.modelPath || this.get('_modelPath');
+    const currentPage = options.startingPage === undefined ? 0 : options.startingPage-1;
+    const perPage = options.perPage || 25;
+
+    // check if user passed in param w/ infinityModel, else check if defined on the route (for backwards compat), else default
+    const perPageParam = typeOfCheck(options.perPageParam, get(this, 'perPageParam'), 'per_page');
+    const pageParam = typeOfCheck(options.pageParam, get(this, 'pageParam'), 'page');
+    const totalPagesParam = options.totalPagesParam || 'meta.total_pages';
 
     delete options.startingPage;
     delete options.perPage;
-    delete options.modelPath;
-
-    this.setProperties({
-      currentPage: startingPage,
-      _firstPageLoaded: false,
-      _perPage: perPage,
-      _modelPath: modelPath,
-      _extraParams: options
+    delete options.perPageParam;
+    delete options.pageParam;
+    delete options.totalPagesParam;
+
+    const infinityModel = InfinityModel.create({
+      currentPage,
+      perPage,
+      perPageParam,
+      pageParam,
+      totalPagesParam,
+      _infinityModelName: modelName,
+      extraParams: options,
+      content: A()
     });
```

Comment:
Part of the motivation for this work is to expose some of the internals of Ember Infinity to people who'd like to extend it.

Therefore - I think we should a) expose this constructor to people to extend in their own apps, and b) allow them to pass a class into the `infinityModel` that should be extended.  Ie:

```
// addon/mixins/route.js

import InfinityModel from 'ember-infinity/lib/infinity-model';
 
infinityModel(modelName, options, boundParams, infinityModelClass=InfinityModel) {
  //...snip
  const infinityModel = infinityModelClass.create({ ...params })
}
```

Then in a users host app they can do:

```
import InfinityModel from 'ember-infinity/lib/infinity-model';
const CustomInfinityModel = InfinityModel.extend({
  //..snip
})

export default Route.extend({
  model() {
    return this.infinityModel('product, { perPage: 10, startingPage: 1 }, {}, CustomInfinityModel);
  }
})
```